INTRODUCTION.md

```diff
@@ -1132,6 +1133,37 @@ A DNS record containing all broker address can thus be used to provide a
 reliable bootstrap broker.
 
 
+#### SSL
+
+If the client is to connect to a broker's SSL endpoints/listeners the client
+needs to be configured with `security.protocol=SSL` for just SSL transport or
+`security.protocol=SASL_SSL` for SASL authentication and SSL transport.
+The client will try to verify the broker's certificate by checking the
+CA root certificates, if the broker's certificate can't be verified
+the connection is closed (and retried). This is to protect the client
+from connecting to rogue brokers.
+
+The CA root certificate defaults are system specific:
+ * On Linux, Mac OSX, and other Unix-like system the OpenSSL default
+   CA path will be used, also called the OPENSSLDIR,  which is typically
+   `/usr/lib/ssl` (on Linux, typcially in the `ca-certificates` package) and
+   `/usr/local/etc/openssl` on Mac OSX (Homebrew).
+ * On Windows the Root certificate store is used.
+
+If the system-provided default CA root certificates are not sufficient to
+verify the broker's certificate, such as when a self-signed certificate
+or a local CA authority is used, the CA certificate must be specified
+explicitly so that the client can find it.
+This can be done either by providing a PEM file (e.g., `cacert.pem`)
+as the `ssl.ca.location` configuration property, or by passing an in-memory
+PEM, X.509/DER or PKCS#12 certificate to `rd_kafka_conf_set_ssl_cert()`.
```

Comment:
nit: they are DER/PEM encoded X.509 certificates and PKCS#12 is an archive file format primarily used to bind the private key with the X/509 certificate. 

It may be nice to mention common extensions as well; .crt, .cer, .p12, .pfx

I think it's worth mentioning mtls as well  as this is where the private key comes into play and where pkcs#12 shines.

Reply:
Thanks for the input, but I'm not comfortable to write that out, feel free to improve this in a future PR :+1: